### PR TITLE
Add long error explanation for E0493

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of #61137.

### CODE REVIEW

File: src/librustc_mir/error_codes.rs [hunk 1]
```diff
@@ -1128,6 +1128,62 @@ Remember this solution is unsafe! You will have to ensure that accesses to the
...
+
+struct Foo {
+    field1: DropType,
+}
+
+static FOO: Foo = Foo { field1: DropType::A }; // We initialize all fields
+                                               // by hand.
+```
+
+And by using `lazy_static`:
```
Comment (Centril) [inline src/librustc_mir/error_codes.rs]: I'm not sure we should (only?) suggest `lazy_static` here. It feels like playing favorites while crates like https://docs.rs/once_cell/1.1.0/once_cell/ might be more idiomatic Rust these days.

File: src/librustc_mir/error_codes.rs [hunk 2]
```diff
@@ -1128,6 +1128,62 @@ Remember this solution is unsafe! You will have to ensure that accesses to the
 cell are synchronized.
 "##,
 
+E0493: r##"
+A type with a `Drop` implementation was desconstructed when trying to build a
+static type.
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/librustc_mir/error_codes.rs]: How do you call a static variable?

File: src/librustc_mir/error_codes.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply src/librustc_mir/error_codes.rs]: Maybe I should just not talk about those crate at all. I'll remove this part.

**Comment by Centril** [comment]
Failed in https://github.com/rust-lang/rust/pull/64427#issuecomment-531286433, @bors r-

Please run `--compare-mode=nll`.

File: src/librustc_mir/error_codes.rs [hunk 3]
```diff
@@ -1128,6 +1128,51 @@ Remember this solution is unsafe! You will have to ensure that accesses to the
 cell are synchronized.
 "##,
 
+E0493: r##"
+A type with a `Drop` implementation was deconstructed when trying to initialize
```
Comment (Centril) [inline src/librustc_mir/error_codes.rs]:
```suggestion
A type with a `Drop` implementation was destructured when trying to initialize
```

**Comment by Mark-Simulacrum** [comment]
Awkwardly stealing this PR for some initial testing... 

@rust-timer queue

**Comment by Alexendoo** [comment]
Ping from triage, any updates? @GuillaumeGomez

**Comment by Centril** [comment]
@GuillaumeGomez updates? // triage

**Comment by GuillaumeGomez** [comment]
Updated!

**Pull request merged**